tests/mmm-ext-fulfill.spec.ts

```diff
@@ -0,0 +1,925 @@
+        );
+        assert.equal(
+          referralBalance,
+          initReferralBalance + expectedBuyPrices.takerFeePaid.toNumber(),
+        );
+        initReferralBalance = referralBalance;
+      }
+
+      let poolAccountInfo = await program.account.pool.fetch(poolData.poolKey);
+      assertIsBetween(
+        poolAccountInfo.spotPrice.toNumber(),
+        LAMPORTS_PER_SOL / 1.03,
+        PRICE_ERROR_RANGE,
+      );
+      assert.equal(
+        poolAccountInfo.lpFeeEarned.toNumber(),
+        expectedBuyPrices.lpFeePaid.toNumber(),
+      );
+      assert.equal(poolAccountInfo.sellsideAssetAmount.toNumber(), 2);
+
+      const buyerNftAtaAddress = await getAssociatedTokenAddress(
+        mint,
+        buyer.publicKey,
+        true,
+        TOKEN_2022_PROGRAM_ID,
+      );
+
+      {
+        const tx = await program.methods
+          .extFulfillSell({
+            assetAmount: new anchor.BN(1),
+            maxPaymentAmount: new anchor.BN(
+              LAMPORTS_PER_SOL + expectedTakerFees + expectedLpFees,
+            ),
+            buysideCreatorRoyaltyBp: 0,
+            allowlistAux: '',
+            takerFeeBp: 100,
+            makerFeeBp: 100,
+          })
+          .accountsStrict({
+            payer: buyer.publicKey,
+            owner: wallet.publicKey,
+            cosigner: cosigner.publicKey,
+            referral: poolData.referral.publicKey,
+            pool: poolData.poolKey,
+            buysideSolEscrowAccount: solEscrowKey,
+            assetMint: mint,
+            sellsideEscrowTokenAccount: poolAta,
+            payerAssetAccount: buyerNftAtaAddress,
+            sellState: sellState,
+            systemProgram: SystemProgram.programId,
+            tokenProgram: TOKEN_2022_PROGRAM_ID,
+            associatedTokenProgram: ASSOCIATED_TOKEN_PROGRAM_ID,
+          })
+          .transaction();
+
+        const blockhashData = await connection.getLatestBlockhash();
+        tx.feePayer = buyer.publicKey;
+        tx.recentBlockhash = blockhashData.blockhash;
+        tx.partialSign(cosigner, buyer);
+
+        await sendAndAssertTx(connection, tx, blockhashData, false);
+      }
+
+      {
+        const expectedMakerFees = LAMPORTS_PER_SOL * 0.01;
+        const expectedReferralFees = expectedTakerFees + expectedMakerFees;
+        const [buyerBalance, referralBalance, buyerAta] = await Promise.all([
+          connection.getBalance(buyer.publicKey),
+          connection.getBalance(poolData.referral.publicKey),
+          getTokenAccount2022(
+            connection,
+            buyerNftAtaAddress,
+            TOKEN_2022_PROGRAM_ID,
+          ),
+        ]);
+
+        assertIsBetween(
+          buyerBalance,
+          initBuyerBalance -
+            LAMPORTS_PER_SOL -
+            expectedLpFees -
+            expectedTakerFees -
+            expectedTxFees -
+            tokenAccountRent, // no token account rent bc seller ata was closed and pool ata opened
```

Comment:
comment might be incorrect here - we do pay token account rent because the original pool's token account was closed, refunded to the pool owner while the new token account is paid for by the buyer